### PR TITLE
Feature ETO-271: automated deployment & versioning

### PR DESCRIPTION
**Pull request by facumoyanosmf**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          python-version: 3.x
+      - uses: actions/cache@v2
+        with:
+          key: ${{ github.ref }}
+          path: .cache
+      - run: pip install mkdocs-material
+      - run: pip install pillow cairosvg
+      - run: pip install mike
+      - run: pip install mkdocs-glightbox
+      - run: mkdocs gh-deploy --force
```
Comment (valeg-etendo) [inline .github/workflows/ci.yml:27]: @facumoyanosmf The --force is not required

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          python-version: 3.x
+      - uses: actions/cache@v2
+        with:
+          key: ${{ github.ref }}
+          path: .cache
+      - name: Setup Github Config
+        run: |
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+      - run: pip install mkdocs-material
```
Comment (valeg-etendo) [inline .github/workflows/ci.yml:46]: @facumoyanosmf Is it necessary to reinstall the dependencies?

Comment (facumoyanosmf) [reply .github/workflows/ci.yml:46]: @valeg-etendo I tried without installing the dependencies but it told me that it does not recognize the mike command.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
```
Comment (valeg-etendo) [inline .github/workflows/ci.yml:5]: @facumoyanosmf Only the 'main' branch should remain

**Pull request merged**